driver-core/src/main/com/mongodb/ConnectionString.java

```diff
@@ -916,7 +916,7 @@ private MongoCredential createCredentials(final Map<String, List<String>> option
 
         if (credential != null && authMechanismProperties != null) {
             for (String part : authMechanismProperties.split(",")) {
-                String[] mechanismPropertyKeyValue = part.split(":");
+                String[] mechanismPropertyKeyValue = part.split(":", 2);
```

Comment:
This ignores the second `:` (a test passes in a value that decodes to a string that includes `:`).